### PR TITLE
Add followers to readme & update migration script readme

### LINKED ISSUE #4583: REMINDER: Document 1.9 migration scripts
### Description

1.9 is going to have a great amount of migration scripts added. We should very carefully document each of them in the migration script readme before release, which currently only mentions a single script.

### Version

upcoming (Edge)

### Upcoming/master Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by Bassoonian**
Adds the follower's branch to the general readme's list of feature branches.

Also sorts the migration script by major release they were introduced in and adds documentation beyond the initial item ball refactor.

## Issue(s) that this PR fixes
Fixes #4583

## **Discord contact info**
bassoonian

### CODE REVIEW

File: migration_scripts/README.md [hunk 1]
```diff
@@ -17,9 +17,293 @@ python3 migration_scripts/*.py ; #run the migration script
...
+```diff
++=== TRAINER_SAWYER_1 ===
++Name: SAWYER
++Class: Hiker
++Pic: Hiker
++Gender: Male
++Music: Hiker
++Double Battle: No
++AI: Check Bad Move / Try To Faint / Check Viability
++
```
Comment (AlexOn1ine) [inline migration_scripts/README.md:124]: We have mugshots and starting status now. can they be added as a showcase?

File: migration_scripts/README.md [hunk 2]
```diff
@@ -17,9 +17,293 @@ python3 migration_scripts/*.py ; #run the migration script
...
+        .backPicYOffset = 13,
+        .backAnimId = BACK_ANIM_DIP_RIGHT_SIDE,
+        .palette = gMonPalette_Bulbasaur,
+        .shinyPalette = gMonShinyPalette_Bulbasaur,
+        .iconSprite = gMonIcon_Bulbasaur,
+        .iconPalIndex = 4,
+        FOOTPRINT(Bulbasaur)
+        .levelUpLearnset = sBulbasaurLevelUpLearnset,
+        .teachableLearnset = sBulbasaurTeachableLearnset,
++        .eggMoveLearnset = sBulbasaurEggMoveLearnset,
```
Comment (AsparagusEduardo) [inline migration_scripts/README.md:297]:
```suggestion
+       .eggMoveLearnset = sBulbasaurEggMoveLearnset,
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Small things to address. Looks good otherwise :)

File: migration_scripts/README.md (hunk 1 above)
Comment (Bassoonian) [reply migration_scripts/README.md:124]: I don't think the migration script is the right place to showcase it, it just shows what changes you can expect for a generic entry. We should probably document the .party file format, but that's out of scope for this PR

Comment (mrgriffin) [reply migration_scripts/README.md:124]: You could pop them in `test/battle/trainer_control.party`, that's usually where I point people. I don't think we have any tests of the trainer data, just the party data, but they'd be easy to write and would help guarantee that `trainerproc` is producing the correct data :)

File: README.md [hunk 3]
```diff
@@ -141,6 +141,13 @@ Based off RHH's pokeemerald-expansion v1.8.0 https://github.com/rh-hideout/pokee
     - [Nature Colors](https://github.com/DizzyEggg/pokeemerald/tree/nature_color) in summary screen by @DizzyEggg
     - [Dynamic Multichoice](https://github.com/SBird1337/pokeemerald/tree/feature/dynmulti) by @SBird1337
     - [Saveblock Cleansing](https://github.com/ghoulslash/pokeemerald/tree/saveblock) by @ghoulslash
+    - [Followers & Expanded IDs](https://github.com/aarant/pokeemerald/tree/followers-expanded-id) by @aarant
+        - May be disabled.
+        - Includes Pokémon followers like in HGSS, including interactions.
+        - Expands the amount of possible object event IDs beyond 255.
+        - Includes an implementation of dynamic overworld palettes (DOWP).
+        - **Additional features**:
+            - *Pokémon overworld sprites up to Generation 8.*
```
[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply README.md:150]: Just made it up right now! 😂 
We generally called it the "Pokémon Debug menu" but I kept thinking that it didn't describe what it did, so probably it's something to discuss on the server if everyone else is ok with the name.
If so, I can update references to it in the repo for the PR that lets you access that menu from the OW Debug Menu.

[the remark this answers is not in the document]
Comment (Bassoonian) [reply README.md:150]: I totally dig the name, but we should advertise its existence and how to open it. Almost no one knows it, and I agree that having "debug" in its name makes it very easy for people to confuse it with the regular debug menu.

File: migration_scripts/README.md (hunk 1 above)
Comment (Bassoonian) [reply migration_scripts/README.md:124]: Totally fine by me, but out of scope for this particular PR :)

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**